# HG changeset patch
# User Rik <rik@octave.org>
# Date 1354823662 28800
#      Thu Dec 06 11:54:22 2012 -0800
# Node ID 71dfd53ed47a0b101eb70648890d13a416bd7d27
# Parent  6faa01ff29678f8a231d50f4a2ac17c3ddd4107f
build: Add rules to clean built files in java/scripts directory.

* scripts/Makefile.am: Add distlean-local rule for java_JAVA_IMAGES.

* scripts/java/module.mk: Add files to CLEANFILES and DISTCLEANFILES variables.

diff --git a/scripts/Makefile.am b/scripts/Makefile.am
--- a/scripts/Makefile.am
+++ b/scripts/Makefile.am
@@ -19,16 +19,20 @@
 # <http://www.gnu.org/licenses/>.
 
 include $(top_srcdir)/build-aux/common.mk
 
 AUTOMAKE_OPTIONS = subdir-objects
 
 EXTRA_DIST =
 
+CLEANFILES =
+
+DISTCLEANFILES =
+
 FCN_FILE_DIRS =
 
 FCN_FILES =
 
 GEN_FCN_FILES =
 
 PKG_ADD_FILES =
 
@@ -446,15 +450,20 @@ EXTRA_DIST += \
   $(IMAGES) \
   $(FCN_FILES) \
   $(FCN_FILES_IN) \
   $(GEN_FCN_FILES) \
   DOCSTRINGS \
   mkdoc.pl \
   mk-pkg-add
 
-DISTCLEANFILES = \
+DISTCLEANFILES += \
   .DOCSTRINGS \
   DOCSTRINGS \
   $(PKG_ADD_FILES) \
   $(DIRSTAMP_FILES) \
   $(GEN_FCN_FILES)
 
+distclean-local:
+	if [ "x$(srcdir)" != "x." ]; then \
+	  rm $(java_JAVA_IMAGES); \
+	fi
+	
diff --git a/scripts/java/module.mk b/scripts/java/module.mk
--- a/scripts/java/module.mk
+++ b/scripts/java/module.mk
@@ -1,12 +1,10 @@
 FCN_FILE_DIRS += java
 
-EXTRA_DIST += java/octave.jar
-
 java_FCN_FILES = \
   java/dlgtest.m \
   java/errordlg.m \
   java/helpdlg.m \
   java/inputdlg.m \
   java/java.m \
   java/javaArray.m \
   java/javaaddpath.m \
@@ -51,28 +49,31 @@ JAVA_IMAGES = \
   $(org_octave_dir)/images/octave.png
 
 java_JAVA_SRC = $(addprefix java/, $(JAVA_SRC))
 
 java_JAVA_CLASSES = $(addprefix java/, $(JAVA_CLASSES))
 
 java_JAVA_IMAGES = $(addprefix java/, $(JAVA_IMAGES))
 
-java_JAVA_IMAGES_src = $(addprefix $(srcdir)/java/, $(JAVA_IMAGES))
-
-java_JAVA_CLASSES_src = $(addprefix $(srcdir)/java/, $(java_JAVA_CLASSES))
+srcdir_java_JAVA_IMAGES = $(addprefix $(srcdir)/java/, $(JAVA_IMAGES))
 
 %.class : %.java
 	$(MKDIR_P) java/$(org_octave_dir)
 	( cd $(srcdir)/java; $(JAVAC) -source 1.3 -target 1.3 -d $(abs_builddir)/java $(org_octave_dir)/$(<F) )
 
-java/images.stamp: $(java_JAVA_IMAGES_src)
+java/images.stamp: $(srcdir_java_JAVA_IMAGES)
 	if [ "x$(srcdir)" != "x." ]; then \
 	  $(MKDIR_P) java/$(org_octave_dir)/images; \
-	  cp $(java_JAVA_IMAGES_src) java/$(org_octave_dir)/images; \
+	  cp $(srcdir_java_JAVA_IMAGES) java/$(org_octave_dir)/images; \
 	fi
 	touch $@
 
 java/octave.jar: java/images.stamp $(java_JAVA_CLASSES)
 	( cd java; $(JAR) cf octave.jar.t $(JAVA_CLASSES) $(JAVA_IMAGES) )
 	mv $@.t $@
 
-EXTRA_DIST += $(java_JAVA_SRC) $(java_JAVA_IMAGES)
+EXTRA_DIST += $(JAR_FILES) $(java_JAVA_SRC) $(java_JAVA_IMAGES)
+
+CLEANFILES += $(JAR_FILES) $(java_JAVA_CLASSES)
+
+DISTCLEANFILES += java/images.stamp
+
